modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/ExpressionFactoryImpl.java

```diff
@@ -487,18 +534,13 @@ private Scalar compile(List<RexNode> nodes, RelDataType type, boolean biInParams
 
         RexProgramBuilder programBuilder = new RexProgramBuilder(type, rexBuilder);
 
-        BitSet unspecifiedValues = new BitSet(nodes.size());
-
         for (int i = 0; i < nodes.size(); i++) {
             RexNode node = nodes.get(i);
 
             if (node != null) {
                 programBuilder.addProject(node, null);
             } else {
-                unspecifiedValues.set(i);
-
-                programBuilder.addProject(rexBuilder.makeNullLiteral(type == emptyType
-                        ? nullType : type.getFieldList().get(i).getType()), null);
+                assert false : "unexpected nullable node";
```

Comment:
This conditional branching is redundant now. I suggest to remove it
```
assert node != null : "unexpected nullable node";

programBuilder.addProject(node, null);
```

Reply:
this method is called from many places, not only from bound and there is no need to process null`s here, so i prefer to stay with this check.

Reply:
>  there is no need to process null`s here, 

Why, then, are you processing them in the conditional branch?
```
            if (node != null) {
                programBuilder.addProject(node, null);
            } else {
                assert false : "unexpected nullable node";
            }
```
> so I prefer to stay with this check

With what check - `else assert false`?

Reply:
My proposal about code cleanup is to replace for-loop with the following:
```
        for (RexNode node : nodes) {
            assert node != null : "unexpected nullable node";
            
            programBuilder.addProject(node, null);
        }
```

Reply:
Perhaps I wasn't clear enough.
I'm just suggesting to replace this code block
```
        for (int i = 0; i < nodes.size(); i++) {
            RexNode node = nodes.get(i);

            if (node != null) {
                programBuilder.addProject(node, null);
            } else {
                assert false : "unexpected nullable node";
            }
        }
```
with this one
```
        for (RexNode node : nodes) {
            assert node != null : "unexpected nullable node";
            
            programBuilder.addProject(node, null);
        }
```

Why you don't like it?)
What fundamental differences do you see with the current behavior?